packages/perseus-editor/src/components/sortable.tsx

```diff
@@ -117,57 +142,75 @@ const SortableArea = createReactClass({
             dragItem.removeEventListener("dragstart", this._listenEvent);
             dragItem.removeEventListener("drop", this._cancelEvent);
         }
-    },
-    render: function () {
-        const sortables = this.state.components.map((component, index) => (
+    }
+
+    render() {
+        const sortables = this.props.components.map((component, index) => (
             <SortableItem
                 index={index}
                 component={component}
                 area={this}
                 key={component.key}
                 draggable={component.props.draggable}
-                dragging={index === this.state.dragging}
+                dragging={index === this.dragging}
             />
         ));
         return (
             <ol className={this.props.className} style={this.props.style}>
                 {sortables}
             </ol>
         );
-    },
-});
+    }
+}
+
+type ItemProps = {
+    area: any;
+    component: any;
+    dragging: boolean;
+    draggable: boolean;
+    index: number;
+};
+
+type DefaultItemProps = {
+    area: ItemProps["area"];
+    component: ItemProps["component"];
+    dragging: ItemProps["dragging"];
+    draggable: ItemProps["draggable"];
+    index: ItemProps["index"];
+};
 
 // An individual sortable item
-const SortableItem = createReactClass({
-    propTypes: {
-        area: PT.shape({
-            onDragEnter: PT.func.isRequired,
-            onDragStart: PT.func.isRequired,
-            onDrop: PT.func.isRequired,
-        }),
-        component: PT.node.isRequired,
-        dragging: PT.bool.isRequired,
-        draggable: PT.bool.isRequired,
-        index: PT.number.isRequired,
-    },
-    handleDragStart: function (e) {
+export class SortableItem extends React.Component<ItemProps> {
+    static defaultProps: DefaultItemProps = {
+        area: "",
```

Comment:
I think it'd be better to not provide all these defaults. The `Sortable` that renders these items looks like it provides all of these props. By making defaults for these props, we make the prop optional, and force this component to have a good default.  In the case of `area`, the default is of the wrong type as it should be an object that provides three callback functions: `onDragEnter`, `onDragStart`, and `onDrop`.